index.d.ts

```diff
@@ -59,6 +59,6 @@ export declare const koaV1Middleware: (
 
 export declare const hapiPlugin: IHapiPlugin<IOptions>
 
-export declare const runWithId: (fn: Function, id?: any) => any
+export declare function runWithId<T>(fn: () => T, id?: string): T
```

Comment:
`fn` may have arbitrary arguments. Not sure if it should be described in `() => T` somehow. WDYT?

Reply:
You don't call the function passed with any arguments though, do you?

Reply:
Nope, I was thinking of functions with optional arguments. After some thinking, it's probably a rare case and it's OK to require a no-args function here. So, I'm resolving this comment.

Reply:
if the argument are optional, that's still assignable to a function without args


```ts
const func: () => void = (id?: string) => {
  if (id) {
    console.log('id');
  }
};

func();
```